### DISCUSSION TITLE
The list-user-repositories API (V3) doesn't return all the repositories

### DISCUSSION THREAD

**Discussion by prahladyeri** [community/community#24396] [Apps, API and Webhooks]
<p>I’m trying to use <a href="https://developer.github.com/v3/repos/#list-user-repositories" rel="nofollow noopener">/v3/repos/#list-user-repositories </a>in one of my open source projects, <a href="https://github.com/prahladyeri/gh_announce" rel="nofollow noopener">prahladyeri/gh_announce</a>. The problem is that this API doesn’t seem to return all the repos a user has, but only the first 30! And I couldn’t find this behavior documented anywhere.</p>
<p>Here is the <a href="https://github.com/prahladyeri/gh_announce/blob/6ce8af4af831bc5b9b5bfc16925c980dd1405062/gh_announce/gh_announce.py#L45" rel="nofollow noopener">offending part of the code</a>:</p>
<pre><code>def do_refresh_tags():
	try:
		# GET /repos/:owner/:repo/topics
		url = "https://api.github.com/users/%s/repos" % config['github_username']
		#print("URL: ",url)
		resp = requests.get(url) #headers={'Accept':'application/vnd.github.mercy-preview+json'}
		data = json.loads(resp.text)
		print("%d repos fetched" % len(data))
		#print("data: ",data)
		if not 'topics' in config: config['topics'] = {}
		for repo in data:
			print("refreshing tags for %s..." % repo['full_name'])
			if not dry_run:
				topics = fetch_topics(repo['full_name'])
				print("%d tags fetched" % len(topics))
				config['topics'][repo['full_name']] = topics
				cfgsaver.save(pkg_name, config)
				print("")
	except Exception as ex:
		print("Error occurred: " + str(ex))
		#raise
</code></pre>
<p>The API URL I’m trying to fetch data from is <a href="https://api.github.com/users/" class="inline-onebox-loading" rel="nofollow noopener">https://api.github.com/users/</a>&lt;username&gt;/repos which is <a href="https://developer.github.com/v3/repos/#list-user-repositories" rel="nofollow noopener">exactly as documented here</a>. Why then, the above call returns only 30 repos when I try with my own user as example, when I have <a href="https://github.com/prahladyeri?tab=repositories" rel="nofollow noopener">34 repositories</a> in total? Is this a limitation of the API?</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/prahladyeri">@prahladyeri</a>,</p>
<p>Thanks for being here! It’s most likely that you are not seeing all your repositories due to pagination. Most calls only return 30 records at a time, this can be increased to 100 via a <code>per_page</code> attribute, but if you have more than 100 you will need to request every page and store the whole list.</p>
<p>Give <a href="http://developer.github.com/v3/#pagination" rel="nofollow noopener">http://developer.github.com/v3/#pagination</a> a read through and see if retrieving subsequent pages works out. I hope this helps!</p>

**Reply by prahladyeri** [reply]
Thanks for the response, @AndreaGriffiths11! I seem to have forgotten to test this for a long time until today when I needed this feature all of a sudden. Thanks again and apologies for not responding earlier.
